File: backend/src/main/java/com/fitness/fitnessaicoach/domain/WorkoutSession.java
package com.fitness.fitnessaicoach.domain;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.UUID;

@Entity
@Table(name = "workout_sessions")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class WorkoutSession {

    @Id
    @GeneratedValue
    private UUID id;

    @ManyToOne(optional = false)
    @JoinColumn(name = "daily_log_id", nullable = false)
    private DailyLog dailyLog;

    @ManyToOne(optional = false)
    @JoinColumn(name = "exercise_id", nullable = false)
    private Exercise exercise;

    private Integer sets;

    private Integer reps;

    private Integer duration;

    private Double caloriesBurned;
}

File: android/fitness-ai-coach-app/app/src/main/java/com/fitness/fitnessaicoach/domain/model/WorkoutSession.kt
package com.fitness.fitnessaicoach.domain.model

data class WorkoutSession(
    val id: String,
    val dailyLogId: String,
    val exerciseId: String,
    val sets: Int?,
    val reps: Int?,
    val duration: Int?,
    val caloriesBurned: Double?
)
